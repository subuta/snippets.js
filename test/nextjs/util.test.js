/* global expect, describe, it */

import { build, format, snippets as s } from 'bld.js'

import next from 'lib/nextjs/util'

describe('nextjs/hoc/connext', () => {
  it('should create util for next.js', () => {
    expect(format(next())).toMatchSnapshot()
  })
})
