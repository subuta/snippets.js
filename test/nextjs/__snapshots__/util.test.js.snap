// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nextjs/hoc/connext should create util for next.js 1`] = `
"import _ from 'lodash'

export const inject = (ctx, key, value) => {
  if (!ctx) return
  if (!_.get(ctx, 'res.locals')) {
    _.set(ctx, 'res.locals', {})
  }
  // mutate ctx and set reference of value.
  ctx.res.locals[key] = value
}

export const extract = (ctx, key) => {
  if (!ctx || !ctx.res) return
  return _.get(ctx, ['res', 'locals', key])
}
"
`;
